pkg/skaffold/runner/dev.go

```diff
@@ -103,19 +104,22 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *kuber
 			instrumentation.AddDevIteration("build")
 			meterUpdated = true
 		}
-		bRes, err := r.Build(ctx, out, r.changeSet.needsRebuild)
+		_, err := r.Build(ctx, out, r.changeSet.needsRebuild)
```

Comment:
```suggestion
		bRes, err := r.Build(ctx, out, r.changeSet.needsRebuild)
```

Reply:
define `var bRes []build.Artifacts` outside the `if` block so that it's used in test.

Reply:
bRes is not used by test.

Reply:
you're using `r.builds` in `r.Test` function. That'll trigger tests for all artifacts. `bRes` has the actual list of artifacts that were rebuilt in the current devloop and so you should pass `bRes` and not `r.builds`

Reply:
Updated.